### PR TITLE
Replace deprecated DLL system calls

### LINKED ISSUE #2294: z/OS DLL open/close/query system calls are deprecated
It looks like the `dllqueryfn()` system call used in [port/zos390/omrsl.c](https://github.com/eclipse/omr/blob/master/port/zos390/omrsl.c) is deprecated https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/dllqfn.htm

We can consider using `dlsym()`, supported since zOS V1R6. 

Similarly, `dllLoad()` and `dllfree()` can be relplaced with `dlopen()` and `dlclose()`.

### PR DESCRIPTION
**Pull request by AidanHa**
Replace `dllqueryfn()`, `dllLoad()` and `dllfree()` with `dlsym()`, `dlopen()` and `dlclose()`

Fixes #2294

### CODE REVIEW

**Comment by rwy0717** [comment]
@genie-omr build zos

File: port/zos390/omrsl.c [hunk 1]
```diff
@@ -75,20 +76,14 @@ omrsl_open_shared_library(struct OMRPortLibrary *portLibrary, char *name, uintpt
 
 	Trc_PRT_sl_open_shared_library_Event1(openName);
 
-	/* dllload() can open a handle to the executable with its name itself (unlike NULL
-	 * on other platforms) provided the executable is itself built as a DLL (in XPLINK
-	 * mode), which anyway is required for enabling symbol resolution.
-	 */
-	handle = dllload(openName);
+	/* 0x002 = RTLD_NOW */
+	handle = dlopen(openName, 0x002);
```
Comment (rwy0717) [inline port/zos390/omrsl.c]: Is `RTLD_NOW` not defined by `<dlfcn.h>`?

Comment (AidanHa) [reply port/zos390/omrsl.c]: I thought so too, but I ran a build on zos, and it didn't recognize it...

**Comment by rwy0717** [comment]
@genie-omr build zos

**Comment by rwy0717** [comment]
@genie-omr build zos

**Comment by rwy0717** [comment]
Last push was just updating comments, no need to rerun CI.

**Pull request merged**